model/Index/Specification/ItemResourceSpecification.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoAdvancedSearch\model\Index\Specification;
+
+use core_kernel_classes_Resource;
+use oat\tao\model\TaoOntology;
+
+class ItemResourceSpecification
+{
+    public function isSatisfiedBy(core_kernel_classes_Resource $resource): bool
+    {
+        $rootClass = $resource->getModel()->getClass(
```

Comment:
Lets check if we need this or we can use the event `ItemUpdatedEvent` to avoid running more queries just to index and Item.

Reply:
There is no `ItemResourceSpecification` anymore, marking this thread as solved.